utils/_booking.py

```diff
@@ -30,10 +30,32 @@ def __hash__(self):
 
 
 class Dataset:
-    def __init__(self, name, ntuples):
+    def __init__(self, name, ntuples, quantities_per_vars):
         self.name = name
         self.ntuples = ntuples
+    def __str__(self):
+        return 'Dataset-{}'.format(self.name)
+
+    def __repr__(self):
+        return self.__str__()
 
+    def add_to_ntuples(self, *new_ntuples):
+        for new_ntuple in new_ntuples:
+            self.ntuples.append(new_ntuple)
+
+    def __eq__(self, other):
+        return self.name == other.name and \
+            self.ntuples == other.ntuples
+
+    def __hash__(self):
+        return hash((
+            self.name, tuple(self.ntuples)))
+
+class DatasetCrown:
```

Comment:
As it us currently implemented, this looks like a complete copy of the Dataset class without any change. Did I miss something?
If it really is a full copy, you could either think of adding a default value of an empty list to the constructor of the dataset class and could you use the same class.
The second possibility, if we really need separate classes, would be to use class inheritance to avoid the code duplication and only reimplement the constructor.